### ISSUE TITLE
Background clicks for spleen deepEdit, adding to segmentation, not removing

### ISSUE THREAD

**Issue by VishalJ99** [Project-MONAI/MONAILabel#1406]
MONAI label version:0.6.0

Slicer version: 5.2.2

Installed sample radiology app, edited deepEdit config file in lib folder to not use pre trained model and set the labels to just spleen and background.

Trained deepEdit model to about 85% accuracy and ran a case. Got following result:
<img width="1117" alt="image" src="https://user-images.githubusercontent.com/51826812/232086030-44d848c1-ec40-4eb2-9dd8-1ec1e6f79c14.png">

**Comment by SachidanandAlle** [comment]
Try deep grow model... It learns better on both foreground and background points... Interactions during training are little different for these 2 models..

**Comment by VishalJ99** [comment]
Am I correct in understanding that the only functional difference between deepEdit and deepGrow is that deepGrow always has background and foreground clicks simulated during training, whereas deepEdit has a probability hyper parameter which determines number of training rounds with simulated clicks.

Are there any other training differences in terms of how the clicks are simulated during training?

**Comment by VishalJ99** [comment]
> 

Am I correct in understanding that the only functional difference between deepEdit and deepGrow is that deepGrow always has background and foreground clicks simulated during training, whereas deepEdit has a probability hyper parameter which determines number of training rounds with simulated clicks.

Are there any other training differences in terms of how the clicks are simulated during training?

**Comment by VishalJ99** [comment]
> Hi @VishalJ99,
> 
> Thanks for reporting this. I can think of two reasons for this:
> 
> 1. From the image I see a blocky segmentation. It seems the DeepEdit model has been trained on low resolution. If you have enough GPU, you may want to increase the image size here: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/deepedit.py#L77
> 2. DeepEdit has a limitation in performance when working on edge segmentation. We are still trying to figure out how to improve this part.
> 
> In the meantime, as @SachidanandAlle suggested, you could also try the DeepGrow model.
> 
> I hope this helps,



Am I correct in understanding that the only functional difference between deepEdit and deepGrow is that deepGrow always has background and foreground clicks simulated during training, whereas deepEdit has a probability hyper parameter which determines number of training rounds with simulated clicks.

Are there any other training differences in terms of how the clicks are simulated during training?

**Comment by SachidanandAlle** [comment]
I suggest going through the training code for them.. u can observe some differences

**Comment by diazandr3s** [comment]
> > Hi @VishalJ99,
> > Thanks for reporting this. I can think of two reasons for this:
> > 
> > 1. From the image I see a blocky segmentation. It seems the DeepEdit model has been trained on low resolution. If you have enough GPU, you may want to increase the image size here: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/deepedit.py#L77
> > 2. DeepEdit has a limitation in performance when working on edge segmentation. We are still trying to figure out how to improve this part.
> > 
> > In the meantime, as @SachidanandAlle suggested, you could also try the DeepGrow model.
> > I hope this helps,
> 
> Am I correct in understanding that the only functional difference between deepEdit and deepGrow is that deepGrow always has background and foreground clicks simulated during training, whereas deepEdit has a probability hyper parameter which determines number of training rounds with simulated clicks.
> 
> Are there any other training differences in terms of how the clicks are simulated during training?

Both DeepGrow and DeepEdit simulate foreground and background clicks during training.
Here are some features that make DeepEdit different from the DeepGrow:

1. DeepEdit allows you to do multilabel segmentation while DeepGrow is more of multi-single-label segmentation. You could segment multiple labels at the same time using DeepEdit, while with DeepGrow you could only segment one label at a time.

2. DeepEdit combines the power of two methods: a non-interactive (i.e. automatic segmentation) and an interactive segmentation method (i.e. DeepGrow), into a single deep learning model. That's why you can see there is a probability hyper-parameter used during training - some iterations are for training the automatic version while others for the interactive model. 

For more information, here is the DeepEdit paper: https://link.springer.com/chapter/10.1007/978-3-031-17027-0_2

You could also check the wiki: https://github.com/Project-MONAI/MONAILabel/wiki/DeepEdit

https://github.com/Project-MONAI/MONAILabel/wiki/Multilabel-DeepEdit

I agree with @SachidanandAlle, the code can also help you to understand the differences.
